proposed/http-message.md

```diff
@@ -776,18 +776,32 @@ interface ServerRequestInterface extends RequestInterface
     public function withQueryParams(array $query);
 
     /**
-     * Retrieve the upload file metadata.
+     * Retrieve the uploaded files.
      *
-     * This method MUST return file upload metadata in the same structure
+     * This method MUST return the file upload metadata in the same structure
      * as PHP's $_FILES superglobal.
      *
      * These values MUST remain immutable over the course of the incoming
      * request. They SHOULD be injected during instantiation, such as from PHP's
      * $_FILES superglobal, but MAY be derived from other sources.
      *
-     * @return array Upload file(s) metadata, if any.
+     * @return UploadedFileInterface[] The uploaded file(s), if any.
```

Comment:
This doesn't cover how to resolve "nested" file uploads per the example in your summary. For instance, if the files were in the form as `example[image]` and `example[icon]`, how would I retrieve these from the returned metadata? What you detail here is an array of upload files, but in your summary, I think this is more likely a _tree_.